### DISCUSSION TITLE
How do you sync a managed user?

### DISCUSSION THREAD

**Discussion by SawkeeReemo** [Taxel/PlexTraktSync#1629] [Q&A] [answered]
I'm brand new to plextraktsync, I have my main account all sync'd up and it's working great. But I'm trying to sync up a managed user separately now, but I can't figure out how to login to it. There was an older thread on here mentioning that the script is supposed to ask you if you want to sync the main account or not, but I am never asked that. I'm using docker container version.

Any help? Thanks!

**Comment by simonc56** [comment, marked as answer]
Just after choosing the server, you're asked which user from this server you want to sync.
You sure you selected the correct server ? (with up/down on your keyboard)
Aloso be aware that managed user is not the same as sharing your library with an other Plex user. Managed users don't have a Plex account for themself.

![image](https://github.com/Taxel/PlexTraktSync/assets/10365746/feba8a97-2393-4abd-8f86-1f3dd47f4b4b)

**Reply by SawkeeReemo** [reply]
Yup. I chose the correct library, but it just spits out an error and quits at that point. Then when I run it again, it works, but only for the main account. 

I did some digging around all day yesterday and figured out how to get around it by “tricking” the setup by manually entering in information in the .env and server.yml files. Seems to have worked. 

Not sure why it crashes at that point, but someone in a Discord server I frequent had the same issue yesterday. I told them to just launch it again and it’ll work. But that was before I knew how to “hack” around.

**Reply by SawkeeReemo** [reply]
Oh! One thing I noticed is that after you add your server, it amends the server.yml with your server info, but does not remove the “default” placeholders. This was showing an “error” in the logs until I removed the default stuff manually from that file. 

I wonder if it somehow not being replaced in there could be tripping it up? (Total shot in the dark here. I have no idea, just noticed it and thought I should note it in case it’s helpful.)

EDIT:
Also, I just realized I didn’t give you specs:
Synology DSM 7.2
plextraktsync in a Docker container
Plex (beta PMS) is running inside DSM, and is not currently in a Docker container

**Reply by simonc56** [reply]
Plex server is read in `.env` file  as PLEX_SERVER=Your_server_name
Then it reads `servers.yml` to get token and urls of the server named accordingly.
Maybe you didn't have PLEX_SERVER in `.env`

**Reply by SawkeeReemo** [reply]
I’ll be honest, I was messing around with this for so long yesterday, that I can’t remember what was there initially. But I do know that when I initially set this up and ran it for the first time, after putting in my credentials and IDs, etc,, when it got to that point it just crashed. But then when I just ran the Docker command again (literally just pushed up in terminal and hit enter), the second time it actually ran, but never asked me to choose a user. 

My work around was to clone the config folder I’m using, [then using info from another post in here](https://github.com/Taxel/PlexTraktSync/discussions/1288), I was able to get things up and running manually.

Not sure if this is a bug or if there needs to be some other specific steps for the containerized version. Maybe it’s Synology being smart/stupid? (Wouldn’t be the first time 😅)

**Comment by AngusNB** [comment]
> figured out how to get around it by “tricking” the setup by _manually entering in information_ in the .env and server.yml files. Seems to have worked.

What did you enter in .env and server.yml to get other users to work?
I would like to add two Managed Users. Ideally I would like to Sync the server Admin and scrobble the other two users.

**Reply by glensc** [reply]
you can run plex-login sub command any time to bring up the login dialogs. managed users are only offered if you choose server the plex account owns.

you can have only one user at a time. you must use a different config directory if you want more than one user.

it may be possible with one config:
1. log in and use main account (not managed user)
2. and turn off username_filter

but then it will track all managed users, and to single trakt account. if you want any other combination, separate config directories are only way to go.

**Reply by AngusNB** [reply]
Thanks, that's what I am finding. 

@SawkeeReemo says he has "tricked" the app. I was just curious as to what he wss referring to.

**Reply by SawkeeReemo** [reply]
Sorry for lag, extremely busy over here IRL. All I did was look at what was in the config files, then I adjusted those values to match what I wanted them to be in the other instances of PlexTraktSync that I set up. I can’t remember off the top of my head exactly what I did, but trust me… if I can figure it out, anyone can. 🤣 It was a matter of getting the user id token for each managed user and replacing the main account’s token with it.. something like that. If I ever get a free moment again, I’ll try to dig in and figure out exactly what I did. (But don’t stay up late waiting… I’m totally buried over here.)

**Reply by AngusNB** [reply]
Ah, ok. I didn't see where you were running multiple instances.

**Reply by SawkeeReemo** [reply]
Yeah, I have a really funky setup. My gf and I watch stuff together on one account, that syncs to Trakt. Then 10 min later another one runs that syncs from Takt to my personal account (updating the watch status), then another to hers. Then we each have our own that go back up to Trakt. This way, no matter if we watch something together or separately, our personal accounts are always correct. Nothing syncs from Trakt to the first account though. That way if she gets ahead of me on a show or something, my personal account watch status remains where I left off. It’s like a big Rube Goldberg device. 😂

**Reply by AngusNB** [reply]
My wife and I each have Plex Managed accounts. I also have the Admin account for shows we watch together. 

I use one Trakt account. Our individual watch is pushed to Trakt. The Admin account syncs all watch history. So the Admin account reflects everything watched in the house but our individual accounts only show what we've watched on our own. 

I use two instances of PlexTraktSync for the two Managed accounts and the old Plex / Trakt plugin for the Admin account.

**Reply by SawkeeReemo** [reply]
Yeah, that’s how I originally set it up, but I ended up with the reverse in my complicated way in order to have the shared account me its own thing, but syncing watched status with the individual accounts. But I also wanted the individual accounts to have their own watch status. 

I have three Trakt accounts and five (!!!) instances of plextraktsync to pull this off. There might be a better way out there, but I’m too burned out to figure it out. 🤣

Would love to be able to do all of this with just one plextraktsync, like it would be awesome to have it all in one place and just run one after the other. There are some other scripts I use where you can tell it to run for each config.yml it has. I wish all of these apps were like that. But also, beggars can’t be choosers! I love that this exists at all!

**Reply by AngusNB** [reply]
Oops, I lied a little. I just looked at my setup.

I am using the old Plugin to Push all 3 user's watched items to Trakt then using PlexTraktSync to do a full Sync for the Admin only. So, only one instance of PlexTraktSync. I was initially going to set it up the way I described earlier. This way was easier.

**Reply by SawkeeReemo** [reply]
Interesting... what is this "old plugin" you are talking about? I wonder if I could streamline my setup at all... I noticed that Trakt had a webhook you can use directly on Plex, but it's only on a paid account. I paid for Trakt for a year to support them, but not sure I will be continuing that. Is that what you were referring to? Or is there something else?

**Reply by AngusNB** [reply]
This plugin. https://github.com/trakt/Plex-Trakt-Scrobbler

Can be configured here: https://trakt-for-plex.github.io/configuration/#/login

There are also configurations options within Plex.

**Reply by SawkeeReemo** [reply]
Oh wow. Looks like it hasn’t been worked on for several years. Amazed it’s still working! I might have to give it a look just for grins.
